src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1097,8 +1100,11 @@ private static string GetConfigurationAndPlatformPropertiesString(ProjectConfigu
         /// </summary>
         private static string GetPropertiesAttributeForDirectMSBuildTask(ProjectConfigurationInSolution projectConfiguration)
         {
-            string directProjectProperties = OpportunisticIntern.InternStringIfPossible(String.Join(";", GetConfigurationAndPlatformPropertiesString(projectConfiguration), SolutionProperties));
-            return directProjectProperties;
+            using RopeBuilder directProjectProperties = ST.GetRopeBuilder();
```

Comment:
Didn't reach StringTools yet, but `RopeBuilder` does not sound descriptive at all. Is it some established pattern / algorithm that I never head about? By its name I assume it takes multiple sequencey things and somehow weaves them together :)

Reply:
On a darker note, it's also indicative of how we feel sometimes when reading the msbuild code base :)

---
In reply to: [486037469](https://github.com/dotnet/msbuild/pull/5663#discussion_r486037469) [](ancestors = 486037469)

Reply:
Haven't looked but guessing it's this: https://en.wikipedia.org/wiki/Rope_(data_structure)

Reply:
Yes, it's a rope, although a simple one since we don't need operations that would make it a tree. We basically just append strings to the end. I want to have the API (informally) reviewed before merging so it is possible that we'll end up with a different name.

Reply:
Renamed to `SpanBasedStringBuilder`.